File: Debugger/combinations.py
string=[]
def comb(given_arr,given_length,temp,index,saved):

    if(len(temp)==given_length and saved==0):
        string.append(temp)
    if index>=len(given_arr):
        return

    comb(given_arr,given_length,temp+[given_arr[index]],index+1,0)
    comb(given_arr, given_length, temp, index + 1,saved+1)
comb([1,2,3,4],2,[],0,0)
print(string)

File: Debugger/combinations.js

var string=""
function arraytostring(array){
    string="";
    for(var i=0;i<array.length;i++){
        string+=array[i]+" ";
    }
    return string;
}
function _combinations(given_arr,given_length,temp,index,saved){
    if(temp.length==given_length && saved==0){
        console.log(temp)
        string+=arraytostring(temp)+"<br>"
    }
    if(index>=given_arr.length){
        return ;
    }
    temp.push(given_arr[index])
    _combinations(given_arr,given_length,temp,index+1,0)
    temp.pop()

    _combinations(given_arr,given_length,temp,index+1,saved+1)
    
}

function combinations(arr,length=undefined){
    if(length==undefined){
        length=arr.length
    }
    string=""
    _combinations(arr,length,[],0,0)
    
}
arr=[1,2,3,4,5]
ans=combinations(arr,2)

console.log(string)